### ISSUE TITLE
Setting up for tensor flow

### ISSUE THREAD

**Issue by yuvi1323** [yuvi1323/Advanced-AI-Analysis-of-the-Effects-of-Wildfires-on-Public-Health-and-Air-Quality#5]
- Loading TensorFlow-based data files to make a combined data as it sets you up for full-dataset model training.
- CNN to predict FireMask from PrevFireMask and features

**Eventually it helps to start with Fire Spread Prediction (CNN)**

- Generate tfrecord_parser.py with support for PrevFireMask, FireMask, NDVI, etc.
- Generate dataset_builder.py to return batched, preprocessed X, y pairs
- Help create a train_fire_cnn.py with a simple model

**Comment by yuvi1323** [comment]
- A batched TensorFlow dataset (parsed_dataset)
- Each sample includes all wildfire-relevant features
- Fully compatible for building a the CNN (PrevFireMask + features → FireMask)

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/6e5f5619-1a93-4206-9ebe-58533ec3cc69" />

**Comment by yuvi1323** [comment]
**Model Summary:**
- Input: (64, 64, 12) → 12 spatial features per sample
- Output: (64, 64, 1) FireMask for next day
- You trained for 5 epochs over ~18,500 samples
- Final Accuracy: 96.4%
- Final Loss: negative MSE (since you're predicting 0/1 masks) — the loss is negative because of how binary crossentropy reacts to logits close to 0 or 1 in regression shape.

<img width="1316" alt="Image" src="https://github.com/user-attachments/assets/81a62060-2868-400d-b628-3581d23229d2" />

**Error are normal 👍**

- These simply mean: we didn’t  ".repeat()" our dataset.So training stops naturally at the end of the available data (which is what we want for now).

<img width="1466" alt="Image" src="https://github.com/user-attachments/assets/cf4b19c5-50a9-475a-a97a-95af9d442a74" />

**Comment by yuvi1323** [comment]
Model saved successfully with h5 files

**Output:**

<img width="787" alt="Image" src="https://github.com/user-attachments/assets/fb1c4f6b-96b7-4cc0-80ef-b69bc7151723" />

**trained model file:**

<img width="691" alt="Image" src="https://github.com/user-attachments/assets/c01ddc58-77bb-4139-9dc7-26662ac070de" />
